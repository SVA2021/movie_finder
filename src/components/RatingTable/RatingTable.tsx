import {Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow} from "@mui/material";
import {FC, memo} from 'react';
import {TFullCard} from "../../features/movieFinder/movieFinderTypes";

interface RatingTableProps {
  movie: TFullCard | null
}

export const RatingTable: FC<RatingTableProps> = memo(({movie}) => {

  function createData(
    name: string,
    rating: string,
    counts: string,
  ) {
    return {name, ratingSummary: `${rating} / ${counts}`};
  }

  const rows = [
    createData('Kinopoisk', `${movie?.ratingKinopoisk ?? ' - '}`, `${movie?.ratingKinopoiskVoteCount ?? ' - '}`),
    createData('Imdb', `${movie?.ratingImdb ?? ' - '}`, `${movie?.ratingImdbVoteCount ?? ' - '}`),
    createData('Await', `${movie?.ratingAwait ?? ' - '}`, `${movie?.ratingAwaitCount ?? ' - '}`),
    createData('RfCritics', `${movie?.ratingRfCritics ?? ' - '}`, `${movie?.ratingRfCriticsVoteCount ?? ' - '}`),
    createData('FilmCritics', `${movie?.ratingFilmCritics ?? ' - '}`, `${movie?.ratingFilmCriticsVoteCount ?? ' - '}`),
    createData('GoodReview', `${movie?.ratingGoodReview ?? ' - '}`, `${movie?.ratingGoodReviewVoteCount ?? ' - '}`),
    {name: 'Количество рецензий', ratingSummary: movie?.reviewsCount ?? ' - '},
  ]

  return (
    <TableContainer component={Paper} data-testid={'rating-table'} >
      <Table aria-label="Рейтинги" sx={{maxWidth: '100%'}} >
        <TableHead>
          <TableRow>
            <TableCell colSpan={2} align={'center'} >Рейтинги</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row) => (
            <TableRow key={row.name} >
              <TableCell component="th" scope="row" align={'left'} padding={'checkbox'} sx={{border: 'none'}} >
                {row.name}
              </TableCell>
              <TableCell align={'center'} padding={'normal'} sx={{border: 'none'}}>
                {row.ratingSummary}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
});