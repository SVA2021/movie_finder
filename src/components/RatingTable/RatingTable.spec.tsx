import {render, screen} from "@testing-library/react";
import {RatingTable} from "./RatingTable";

describe('render RatingTable', () => {

  const setUp = () => render(<RatingTable movie={null} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('rating-table')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/Kinopoisk/i)).toBeInTheDocument();
    expect(screen.getByText(/Imdb/i)).toBeInTheDocument();
    expect(screen.getByText(/Await/i)).toBeInTheDocument();
    expect(screen.getByText(/RfCritics/i)).toBeInTheDocument();
    expect(screen.getByText(/FilmCritics/i)).toBeInTheDocument();
    expect(screen.getByText(/GoodReview/i)).toBeInTheDocument();
    expect(screen.getByText(/Количество рецензий/i)).toBeInTheDocument();
  });

})