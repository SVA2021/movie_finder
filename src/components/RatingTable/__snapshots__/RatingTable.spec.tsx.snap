// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RatingTable snapshot test 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  data-testid="rating-table"
>
  <table
    aria-label="Рейтинги"
    class="MuiTable-root css-fj9a4a-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
          colspan="2"
          scope="col"
        >
          Рейтинги
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          Kinopoisk
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          Imdb
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          Await
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          RfCritics
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          FilmCritics
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          GoodReview
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           -  /  - 
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-rk8gnx-MuiTableCell-root"
          scope="row"
        >
          Количество рецензий
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-iixv6p-MuiTableCell-root"
        >
           - 
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
